src/shipit_pulse_listener/shipit_pulse_listener/listener.py

```diff
@@ -31,47 +33,49 @@ def __init__(self, configuration):
         self.api_url = configuration['phabricator_url']
         self.api_token = configuration['phabricator_token']
 
-        # Start on first page
-        self.before = None
+        # Start by getting top id
+        diffs, _ = self.request_phabricator(limit=1)
+        assert len(diffs) == 1
+        self.top_id = diffs[0]['id']
+
+    def request_phabricator(self, after=None, limit=20, order='newest'):
+        '''
+        Load raw differential objects from the api
+        '''
+        logger.debug('Loading phabricator differentials', after=after, limit=limit, order=order)
+        url = '{}/differential.diff.search'.format(self.api_url)
+        payload = {
+            'api.token': self.api_token,
+            'after': after,
+            'order': order,
+            'limit': limit,
+        }
+        response = requests.post(url, payload)
+        response.raise_for_status()
+        data = response.json()
+        assert data['error_code'] is None, \
+            'Conduit error: {} - {}'.format(
+                data['error_code'],
+                data['error_info'],
+            )
+
+        after = data['result']['cursor']['after']
+        after = after and int(after)
+        return data['result']['data'], after
```

Comment:
Here you could return `data['result']['data'], after is not None`, that is making the second variable a "is the query finished returning results?"